content/6_Advanced/Multiplicative.mdx

```diff
@@ -64,6 +267,335 @@ int main() {
+	fill(is_composite, is_composite + n, false);
+	for (int i = 2; i < n; ++i) {
+		if (!is_composite[i]) { prime.push_back(i); }
+		for (int j = 2; i * j < n; ++j) { is_composite[i * j] = true; }
+	}
+}
+// EndCodeSnip
+
+int main() {
+	int n, a, b;
+	cin >> n >> a >> b;
+	sieve(n);
+
+	int c = 0;
+	while (a * c + b < prime.size()) c++;
+
+	cout << prime.size() << ' ' << c << '\n';
+
+	c = 0;
+	while (a * c + b < prime.size()) { cout << prime[a * c + b] << ' ', c++; }
+	cout << endl;
+}
+```
+
+</CPPSection>
+</LanguageSection>
+
+## Example - Counting Primes
+
+<FocusProblem problem="f4" />
+
+There's two ways to do this problem.
+The first solution has a higher time complexity but a less complex implementation,
+while the second has a lower time complexity at the cost of a more complex implementation.
+
+For a more complete explanation of these algorithms, refer to [this CF blog post](https://codeforces.com/blog/entry/91632).
+
+## Explanation 1
+
+Utilizes a dynamic programming approach based on a recursion relation derived from sieving.
+
+The algorithm iteratively reduces the count of numbers that are not divisible by primes, utilizing a recursive formula.
+It achieves a complexity of $O(N^{3/4}\log^{1/2}N)$.
+
+### Implementation
+
+**Time Complexity: $\mathcal{O}(n^{3/4})$**
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <algorithm>
+#include <cmath>
+#include <iostream>
+#include <vector>
+
+using ll = long long;
+using std::cout;
+using std::endl;
+using std::pair;
+using std::vector;
+
+ll count_primes(ll n) {
+	vector<ll> v;
+	for (ll k = 1; k * k <= n; k++) {
+		v.push_back(n / k);
+		v.push_back(k);
+	}
+	sort(v.begin(), v.end());
+	v.erase(std::unique(v.begin(), v.end()), v.end());
+
+	/*
+	 * return i such that v[i] = x
+	 * since v[i] = i + 1 for i <= sqrt(n) and v[v.size() - i] = n / i for i <=
+	 * sqrt(n), we can calculate index in O(1)
+	 */
+	ll sq = sqrt(n);
+	auto geti = [&](ll x) {
+		if (x <= sq) {
+			return (int)x - 1;
+		} else {
+			return (int)(v.size() - (n / x));
+		}
+	};
+
+	vector<ll> dp(v.size());
+	// S(n, 0) = n
+	for (int i = 0; i < v.size(); i++) { dp[i] = v[i]; }
+
+	int a = 0;
+	for (ll p = 2; p * p <= n; p++) {
+		// this condition is true for primes
+		if (dp[geti(p)] != dp[geti(p - 1)]) {
+			a++;
+			for (int i = (int)v.size() - 1; i >= 0; --i) {
+				if (v[i] < p * p) { break; }
+				dp[i] -= dp[geti(v[i] / p)] - a;
+			}
+		}
+	}
+
+	return dp[geti(n)] - 1;
+}
+
+int main() {
+	ll n;
+	std::cin >> n;
+	cout << count_primes(n) << endl;
+}
+```
+
+</CPPSection>
+</LanguageSection>
+
+## Explanation 2
+
+There exists a faster way to compute primes in O(n^2/3), see [Maksim1744’s Codeforces Blog](https://codeforces.com/blog/entry/91632) for more details. Below is the implementation using the fenwick code snippet from the [PURS module](https://usaco.guide/gold/PURS?lang=cpp#solution---dynamic-range-sum-queries-with-a-bit)
+
+### Implementation
+
+**Time Complexity: $\mathcal{O}(n^{2/3})$**
```

Comment:
there should be a log factor somewhere

Reply:
yeye ur right mb fixed (will commit asap)